### ISSUE TITLE
ctrl-space wont open word recommendation but also blocks them

### ISSUE THREAD

**Issue by ali80** [maestrith/AHK-Studio#134]
I don't think that you've mentioned ctrl+space do this function but since most code editors use it for this purpose, I (and, guess many other users) keep pressing it and not only it doesn't do anything, it also closes the word recommendation if it was already open :), so you have to delete a few characters for the word recommendation to popup again.

**Comment by maestrith** [comment]
Ah, Ctrl+Space is for something else actually, it's for jumping between multiple documents if you have more than one Scintilla control in the GUI.  Alt+W is the hotkey for Word Recommendations.

**Comment by ali80** [comment]
well Alt+W doesn't work either:
type something, when word recommendation pops up, press escape to close it, now Alt+w ... nothing ...
IMO learning shortcuts is part of speeding up your workflow in a program and feeling at home, so it would be much easier to start using a tool that has familiar shortcuts (specially to us shortcut junkies :) )
it would be much appreciated if you could change or maybe set up a new shortcut profile which is familiar with shortcuts used in tools such as sublime or notepad++.

**Comment by maestrith** [comment]
I have no idea what the shortcuts are for those programs and right now I have no means of getting them.

**Comment by maestrith** [comment]
Sorry, It is Ctrl+Q.  I changed mine to Alt+W on my PC

**Comment by ali80** [comment]
I can get you a list of shortcuts if you want.
I also think the idea of shortcut profiles is cool, i've seen some programs have it and I very much liked it.

**Comment by maestrith** [comment]
I am once again sorry, it was Ctrl+W.... I need to learn to read what I type before sending it.
If you would like, I would not object to a hotkey list.  I just have not used enough editors out there before writing Studio to get the feel for them.

**Comment by ali80** [comment]
Ha :) this is the kind of problem when you create you own shortcuts, it causes confusion, there is only a certain amount of shortcuts a human mind can handle!

**Comment by ali80** [comment]
well since ahk-studio has a lot of functionalities similar to sublime here is a list of sublime shortcuts
https://gist.github.com/eteanga/1736542
I myself think that some of your shortcut placements are more logical than sublime (which is very good in this category), but what good would be a shortcut if you can't remember it.

**Comment by ali80** [comment]
if you want the list in a particular format maybe I can save you some time.

**Comment by maestrith** [comment]
I can not promise anything, but I will look at them and see what I can do with them.  I think that I should just remove all shortcuts that are not system specific.

**Comment by ali80** [comment]
well that's OK, I understand.
I don't know what you mean by non system specific shortcuts.

**Comment by maestrith** [comment]
Like Ctrl+W (I think) is close.  I just don't know anymore.... Maybe having a huge list of all the editors in the world and the hotkeys for them and then mapping them to Studio would be helpful...

**Comment by ali80** [comment]
yeah, definitely, even all browser use this shortcut for the very same purpose :)
well actually if you consider the commonly used editors, the list is not that big, here is a list from stackoverflow 2016 survey :
![image](https://user-images.githubusercontent.com/6554605/30797361-987fa0ca-a1e2-11e7-810f-431eed059fe9.png)
1. visual studio, has complex mostly two step shortcuts (I think the style is somewhat similar to emacs)
2. sublime, very good set of shortcuts and has very similar functions with ahk-studio
3. notepad++, widely used but lacks a lot of features of sublime
4. vim, used mostly by Linux gurus, the concepts of shortcuts have gone to the next level in this one (in a few words everything is a shortcut unless you tell it that you want to type, the default program state is the shortcut state)
5. eclipse, too heavy, don't remember its shortcuts but don't think they were anything bizarre.
IMO sublime shortcuts are the most logical and also easiest way to go.
if you have a particular list of functions that need shortcuts for, that would be a good starting point for me, I will also try to provide a list of commonly used windows shortcuts.

**Comment by maestrith** [comment]
Personally I just want everyone to be happy with the program, so I really don't have a specific function in mind.

**Comment by ali80** [comment]
here is a list of my commonly used shortcuts and my comment on current ahk-studio shortcut


Function | Sublime Text | Visual Studio | ahk-studio | Comment
-- | -- | -- | -- | --
delete line | Ctrl+⇧+K | Ctrl+Shift+L | Shift+F2 | bad
New Project | Ctrl+Shift+N | Ctrl+Shift+N | Ctrl+PGDN | bad
comment/un-comment current line(s) | Ctrl+ / | Ctrl+ / (though plugin) | Ctrl+j | bad
show   auto-complete suggestion | Ctrl+Space | Ctrl+Space | Ctrl+W | bad
move line (or selection) up | Ctrl+⇧+↑ | Alt+UP | Ctrl+⇧+↑ | good
move line   (or selection) down | Ctrl+⇧+down | Alt+Down | Ctrl+⇧+down | good
find | Ctrl+F | Ctrl+F | Ctrl+F | good
replace | Ctrl+H | Ctrl+H | Ctrl+H | good
Run File/Project | Ctrl+B | F5 | Alt+R | good
New File | Ctrl+N | Ctrl+N | Ctrl+N | good
Jump Backward | Alt+- | Ctrl+- | Alt+Left | good
Jump Forward | Alt+shift+- | Ctrl+Shift- | Alt+Right | good
next file | Ctrl+PGDWN |   | Ctrl+PGDWN | good
previous   file | Ctrl+PGUP |   | Ctrl+PGUP | good
Menu Search | Ctrl+Shift+P |   | Alt+M | good
goto line in   current file | Ctrl+G | Ctrl+G | Ctrl+G | good
copy current line (if no selection) | Ctrl+C | Ctrl+C |   | if   selection is empty, doesn't do anything
cut current   line (if no selection) | Ctrl+X | Ctrl+X |   | if selection is empty,doesn't do   anything
goto everything |   | Ctrl+T | Alt+Ctrl+O | not   bad, too complex
toggle   bookmark | Ctrl+F2 | Ctrl-K, Ctrl-K |   | not implemented / not   working
find in files | Ctrl+⇧+F | Ctrl+⇧+F |   | not   implemented?
find next | F3 | F3 |   | works in quick find window   but doesn't work in find window
find previous | Shift+F3 | Shift+F3 |   | works in quick find window but doesn't work in find   window
insert line   after | Ctrl+↩ | Ctrl+⇧+↩ |   |  
insert line before | Ctrl+⇧+↩ | Ctrl+↩ |   |  
duplicate   line(s) | Ctrl+⇧+D |   |   |  
goto symbol | Ctrl+R |   |   |  
File Search | Ctrl+P |   | Alt+J |  
next bookmark | F2 | Ctrl-K, Ctrl-N |   |  
previous   bookmark | ⇧+F2 | Ctrl-K, Ctrl-P |   |  
clear bookmarks | Ctrl+⇧+F2 |   |   |  
go to   definition of |   | F12 |   |

**Comment by rediffusion** [comment]
@maestrith 

> Alt+W is the hotkey for Word Recommendations.

FYI for me, it's Ctrl+W how to change to Alt+W?

**Comment by maestrith** [comment]
Alt+M EH {Enter} ;Edit Hotkeys
Enter Words In Document {Enter}
Then change it using the button or I believe Alt+H
